### DISCUSSION TITLE
Problema totale importo nelle righe fattura

### DISCUSSION THREAD

**Discussion by Chrisgo72** [fattureincloud/api#387] [Q&A] [answered]
Buongiorno, ho implementato una soluzione per generare fatture da un gestionale esterno.
Le fatture vengono create utilizzando il gross_price a cui viene spesso abbinato uno sconto percentuale.
Queste fatture vengono scartate quando ci sono righe con quantità maggiore di 1 e con sconto applicato.

Controllando il file xml generato, risulta che lo sconto applicato (che nel xml viene convertito da percentuale ad Euro) viene calcolato come importo intero invece che diviso per le quantità.

Esempio di fattura scartata ieri:

Questa la riga fattura inviata e ricevuta come result (lo sconto è del 44,6556%):
```
{
	"code" : "8014808074783",
	"discount" : 44.6555,
	"discount_highlight" : false,
	"ei_raw" : null,
	"gross_price" : 494.9,
	"net_price" : 405.65574,
	"qty" : 3,
	"vat" : 
	{
		"description" : "",
		"id" : 0,
		"value" : 22
	}
},
```
Questo quanto riportato nel xml scartato

```
<Quantita>3.00</Quantita>
<PrezzoUnitario>405.65573770</PrezzoUnitario>
        <ScontoMaggiorazione>
          <Tipo>SC</Tipo>
          <Importo>543.45</Importo>
        </ScontoMaggiorazione>
<PrezzoTotale>673.52</PrezzoTotale>
```

Lo sconto nel xml dovrebbe essere di 181,15 Euro, non 543,45 
Nel frontend della fattura creata i valori sono tutti corretti, il problema è nel xml generato.

**Comment by emajo** [comment]
Ciao @Chrisgo72, le consiglio di scrivere al nostro supporto clienti in modo da andare ad analizzare più approfonditamente il documento in questione.

Grazie,
Emanuele.

**Reply by valmoz** [reply]
@Chrisgo72 solo per chiarire: lato community non abbiamo la possibilità di accedere ai log delle vostre chiamate API, in quanto per poterlo fare è necessario il company ID; questo campo non può essere passato lato community pubblica, per questo motivo siamo costretti a reindirizzarvi al supporto clienti che invece è privato.
Qui di seguito le modalità per contattare il nostro supporto:
https://developers.fattureincloud.it/docs/support

Grazie
Mauro

**Comment by Chrisgo72** [comment, marked as answer]
Ok, vi ringrazio per la risposta ed ho provveduto a fare richiesta al supporto. Indagando maggiormente ho scoperto che il problema avviene nel caso di righe con quantità maggiore di 1 e sconto percentuale con 3 cifre dopo la virgola. In tal caso l'xml converte lo sconto in importo ma non divide l'importo ottenuto per la quantità.
Quindi
3 pz, importo 494 sconto 50% xml corretto con sconto percentuale del 50%
3 pz importo 494 sconto 44.6555% xml errato con sconto convertito ad importo ma calcolato non correttamente.

**Reply by valmoz** [reply]
Buongiorno @Chrisgo72 
Ok, restiamo in attesa che il supporto ci passi la sua richiesta. Purtroppo visto il periodo dell'anno ci sono molti ticket in attesa, e dovremo aspettare un po'.

Nel frattempo le segnalo che il nostro software effettua arrotondamenti a due cifre, quindi il problema potrebbe essere legato a questo.
Forse questa guida potrebbe esserle utile: [Invoice Totals](https://developers.fattureincloud.it/docs/guides/invoice-totals)

Grazie
Mauro

**Reply by valmoz** [reply]
Buongiorno @Chrisgo72 il supporto ha preso in carico la sua richiesta?

Grazie
Mauro

**Comment by Chrisgo72** [comment]
Buongiorno Mauro,
Le confermo che il supporto ha preso in carica la richiesta.

La ringrazio per l’interessamento.

**Reply by valmoz** [reply]
Perfetto, chiudo qui allora.

Grazie
Mauro
